components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -139,18 +154,276 @@ TEST_CASE ("surface_coupling")
+  auto surf_u_mom_flux_h  = surf_u_mom_flux_f.get_reshaped_view<Real*,Host>();
+  auto surf_v_mom_flux_h  = surf_v_mom_flux_f.get_reshaped_view<Real*,Host>();
+  auto T_mid_h            = T_mid_f.get_reshaped_view<Real**,Host>();
+  auto p_mid_h            = p_mid_f.get_reshaped_view<Real**,Host>();
+  auto z_mid_h            = z_mid_f.get_reshaped_view<Real**,Host>();
+  auto pseudo_density_h   = pseudo_density_f.get_reshaped_view<Real**,Host>();
+  auto horiz_winds_h      = horiz_winds_f.get_reshaped_view<Real***,Host>();
+  auto qv_h               = qv_f.get_reshaped_view<Real**,Host>();
+  auto precip_liq_surf_h  = precip_liq_surf_f.get_reshaped_view<Real*,Host>();
+
+  // Create SC object and set number of import/export fields
+  control::SurfaceCoupling coupler(fm);
+  coupler.set_num_fields(num_total_imports, num_scream_imports, num_exports);
+
+  // Register fields in the coupler. These match the scr_names_x2a/a2x from
+  // scream_cpl_indices.F90. When radiation is fully implemented, RRTMGP
+  // fields will be replaced with correct fields.
+  coupler.register_import("surf_latent_flux", 0);
+  coupler.register_import("surf_sens_flux",   1);
+  coupler.register_import("unused",           2);
+  coupler.register_import("surf_u_mom_flux",  3);
+  coupler.register_import("surf_v_mom_flux",  4);
+  coupler.register_import("RRTMGP",           5);
+  coupler.register_import("RRTMGP",           6);
+  coupler.register_import("RRTMGP",           7);
+  coupler.register_import("RRTMGP",           8);
+  coupler.register_import("RRTMGP",           9);
+  coupler.register_import("unused",           10);
+  coupler.register_import("unused",           11);
+  coupler.register_import("unused",           12);
+  coupler.register_import("unused",           13);
+  coupler.register_import("unused",           14);
+  coupler.register_import("unused",           15);
+  coupler.register_import("RRTMGP",           16);
+  coupler.register_import("unused",           17);
+  coupler.register_import("RRTMGP",           18);
+  coupler.register_import("unused",           19);
+  coupler.register_import("unused",           20);
+
+  coupler.register_export("T_mid",            0);
+  coupler.register_export("Sa_ptem",          1);
+  coupler.register_export("z_mid",            2);
+  coupler.register_export("Sa_u",             3);
+  coupler.register_export("Sa_v",             4);
+  coupler.register_export("p_mid",            5);
+  coupler.register_export("Sa_dens",          6);
+  coupler.register_export("tracers",          7, 0);
+  coupler.register_export("set_zero",         8);
+  coupler.register_export("precip_liq_surf",  9);
+  coupler.register_export("set_zero",         10);
+  coupler.register_export("set_zero",         11);
+  coupler.register_export("set_zero",         12);
+
+  // Complete setup of importer/exporter, providing raw_data
+  double* import_raw_data = new Real[ncols*num_total_imports];
```

Comment:
It looks like you are not deallocating these. Not a huge deal, since the OS will do it, but still...